python/repeated.c

```diff
@@ -187,9 +187,29 @@ PyObject* PyUpb_RepeatedContainer_GetOrCreateWrapper(upb_array* arr,
   return ret;
 }
 
+static PyObject* PyUpb_RepeatedContainer_MergeFrom(PyObject* _self,
+                                                   PyObject* args);
+
+PyObject* PyUpb_RepeatedContainer_DeepCopy(PyObject* _self, PyObject* value) {
+  PyUpb_RepeatedContainer* self = (PyUpb_RepeatedContainer*)_self;
+  PyUpb_RepeatedContainer* clone = (void*)PyType_GenericAlloc(Py_TYPE(_self), 0);
+  const upb_fielddef* f = PyUpb_RepeatedContainer_GetField(self);
+  if (clone == NULL) return NULL;
+  clone->arena = PyUpb_Arena_New();
+  clone->field = (uintptr_t)PyUpb_FieldDescriptor_Get(f);
+  clone->ptr.arr =
+      upb_array_new(PyUpb_Arena_Get(clone->arena), upb_fielddef_type(f));
+  PyUpb_ObjCache_Add(clone->ptr.arr, (PyObject*)clone);
```

Comment:
you cache this, but then might throw it away if the merge from fails.  That feels off to me

Reply:
We have to do it that way, because the destructor will try to remove it from the cache, and assert-fail if it's not present there.

MergeFrom() should basically never fail, so there isn't a significant efficiency concern here.